src/libpmemfile/preload.c

```diff
@@ -1919,46 +2197,89 @@ dispatch_syscall_fd_first(long syscall_number,
 {
 	switch (syscall_number) {
 
-	case SYS_write:
+	case SYS_write: {
+		if (!is_accessible((void *)arg1, (size_t)arg2))
+			return -EFAULT;
+
 		return fd_first_pmemfile_write(arg0, arg1, arg2);
+	}
+
+	case SYS_writev: {
+		int ret;
+		if ((ret = verify_iovec(arg1, arg2)))
+			return ret;
 
-	case SYS_writev:
 		return fd_first_pmemfile_writev(arg0, arg1, arg2);
+	}
+
+	case SYS_read: {
+		if (!is_accessible((void *)arg1, (size_t)arg2))
```

Comment:
And this only checks for the memory region being readable, right?
While read's buffer should also be writable.
Not that I know of any similar easy-to-implement way of checking for that.

Reply:
Yes, it only checks for memory to be readable. But it's enough to handle pjdfstest and LTP tests. It's not supposed to be anything more.